data/src/main/java/io/gnosis/kouban/data/managers/SearchManager.kt

```diff
@@ -77,11 +80,10 @@ data class TransactionTimestampFilter(
 
 ) : BoundaryFilter<Transaction>() {
 
-    override fun apply(item: Transaction): Boolean {
-        return with(item.timestampAsDate()) {
-            (upperBound == null || before(upperBound)) && (lowerBound == null || after(lowerBound))
+    override fun apply(item: Transaction): Boolean =
+        with(item.timestampAsDate().onlyDate()) {
```

Comment:
timestampAsDate().onlyDate() sounds strange. Currently timestampAsDate is used only here. Does it make sence to return only date directly from timestampAsDate()?

Reply:
We need to clear the hours, minutes, seconds and milliseconds of the date in order for 
`equals` to ever work as intended, as we just want to compare dates and not time. This may not be the case everytime, therefore decoupling those 2 methods makes sense.

Reply:
This is unfortunate that the datatype is called date here ... But it js not crucial ... If we want we can call it stripTime()

Reply:
then use https://developer.android.com/reference/kotlin/java/time/LocalDate

Reply:
Yes that would ve possible (also use LocalDateTime, but not sure if this is required right now, because of a method name) ... Guys this PR is open for 2h because of naming discussions ....